### ISSUE TITLE
Improve feature inferencing exception

### ISSUE THREAD

**Issue by woop** [feast-dev/feast#1838]
```
feast.errors.RegistryInferenceFailure: Inference to fill in missing information for DataSource failed. 
                            Unable to infer DataSource event_timestamp_column due to multiple possible columns satisfying
                            the criteria. ^timestamp created_timestamp
                            . Try filling the information explicitly.
```

The exception above (which happens when feature column inferencing fails) is very unfriendly to users. We should be clearer with what went wrong and what the user should do "fill the information" is too vague.

**Comment by achals** [comment]
This has been fixed as of #1790

**Comment by achals** [comment]
- [ ]
